[#44853] moved open-storage button to header

Kharonus opened the pull request:
- https://community.openproject.org/work_packages/44853
- removed action button from file list
- added link button to storage header

Kharonus committed:
[#44853] moved open storage to header

- https://community.openproject.org/work_packages/44853
- removed action button from file list
- added link button to storage header

Kharonus commented:
writing the logic now to the header kind of let me think, if we have a wrong separation in this component.

Instead we could have s.th. like this:

```text
filesTabComponent
  attachmentsComponent (not viewed in detail here ...)
  storageComponent
    storageHeaderComponent
    storageInformationBoxComponent
    storageFileListComponent
    storageActionBarComponent
  storageComponent
    [...]
```

What do you think, @b12f , @apfohl

apfohl commented:
> filesTabComponent
>   attachmentsComponent (not viewed in detail here ...)
>   storageComponent
>     storageHeaderComponent
>     storageInformationBoxComponent
>     storageFileListComponent
>     storageActionBarComponent
>   storageComponent
>     [...]

@Kharonus This is the structure I would assume anyways. Might be worth taking it there. I'm a fan.

b12f commented:
> filesTabComponent
>   attachmentsComponent (not viewed in detail here ...)
>   storageComponent
>     storageHeaderComponent
>     storageInformationBoxComponent
>     storageFileListComponent
>     storageActionBarComponent
>   storageComponent

I think the header and perhaps the action bar are small enough that they can just be part of the storage component.

Kharonus commented:
Oh, yes, definitely. The description was maybe a bit misleading. I didn't think about introducing an own angular component for each of those "components". some will be small and hence inlined into the parents template.

nevertheless, the header currently is one layer above, it is a sibling to the list, while the list contains (!) the box and the action bar.

Alright, I'll use this small increment here to rework a bit of the structure for storages.

Kharonus committed:
[#44853] renamed file-link-list to storage

- file-link-list component renamed to storage component, which is
  container of the list, the header, the info box, and the action bar
- renaming of directories and module

Kharonus commented:
of course this PR now contains some renamings, to summarize:

- the directory structure now features `storages` instead of `file-links`, same for the module those components are in
- the header is now part of the `storage` component (previously `file-link-list`) instead of the `files-tab` component
- and of course: there is now an icon button next to the header leading to the storage url

b12f commented on frontend/src/app/shared/components/storages/storage/storage.component.html:
```diff
@@ -1,6 +1,18 @@
+    <span class="op-tab-content--header-text" [textContent]="storage.name"></span>
+    <button
+      class="op-tab-content--header-action spot-link"
+      [title]="text.openStorage()"
```
this should also be available as an aria attribute, titles are not read by screenreaders afaik

Kharonus replied:
hmm, I thought, on most interaction elements title is interpreted. Checking accessibility tab in firefox gives this:

![image](https://user-images.githubusercontent.com/38206611/200832396-26f0d00b-0cf2-404e-970e-942c7606a584.png)

the title appears within the description. Is there an aria attribute you would add additionally?

b12f replied:
There's a ton of conflicting information surrounding this, mostly because what exactly gets read out loud depends on the screenreader software, OS, and browser. Safest bet is to have both.

b12f replied:
https://dev.opera.com/articles/ux-accessibility-aria-label/#accessible-name-calculation
https://accessible360.com/accessible360-blog/use-aria-label-screen-reader-text/
https://pauljadam.com/demos/title-aria-label.html

Kharonus replied:
so, I could add `aria-label` in addition. From what I was reading now, seems to be the best bet.

b12f reviewed (approved):
lgtm otherwise

Kharonus committed:
[#44853] changed button to a
